policies.go

```diff
@@ -263,9 +263,6 @@ type tokenAwareHostPolicy struct {
 }
 
 func (t *tokenAwareHostPolicy) SetPartitioner(partitioner string) {
-	t.mu.Lock()
```

Comment:
The only thing that SetPartitioner does is call to the fallback policies Partitioner (which in the case of HostPool and RoundRobin are no-ops)  then calls to the resetTokenRing which is where I moved the lock to.